### DISCUSSION TITLE
Jtx suggestions / bugs

### DISCUSSION THREAD

**Discussion by LoveSponge** [TechbeeAT/jtxBoard#670] [Ideas]
First and foremost, nice app! I really enjoy using it to create journal entries, which I never really did before!

The following list is a collection of observations that I think could make the app more useful;

## Widget

- Colour scheme (allow to always use 'light', 'dark' or 'follow system')

## Bugs

- BUG: (known) opening item often brings blank screen
- Unable to navigate cursor around text with the 'hold space bar to move cursor' feature (Samsung keybaord)
- There is a strange issue with filtering where recurring tasks end up not coming up on the task screen if you complete a previous recurrence but the next is on the same day...

## Suggestions

- Link the same task to multiple notes / journals
- Recurring subtask makes allot of entries - try 'every 8 hours for 1 month' (3 entries per day...) maybe limit or allow customisation of a limit?
- Customise task completion sound
- Link task to calendars already on device? (Like task.org)
- Option to show tasks that are linked on task screen
- Option to do 'negative' filtering (NOT 'x' category, NOT Cancelled, etc...)
- Should cancelled tasks be considered 'completed' ? And thus not shown when filtering with 'hide completed tasks'
- Recurring tasks to recur either by when they were due or when they were actually completed
- Recurring tasks show every recurrence for the next year if set to 'never' end... Maybe only show the next 5 or so occurances maximum at any one point...

**Comment by patrickunterwegs** [comment]
Thank you @LoveSponge for your proposals and sorry for the late reply. 
So about your points: 

- Colour scheme (allow to always use 'light', 'dark' or 'follow system') -> I might add this in the future, for now there are some problems with the widget library that need to be solved by the library itself...
- BUG: (known) opening item often brings blank screen -> This should hopefully be solved by now
- Unable to navigate cursor around text with the 'hold space bar to move cursor' feature (Samsung keybaord) -> Regrettably there is some weird keyboard behaviour with the Material3 components that I'm using. But this is coming from the component or the keyboard provider, there's not much I can do about it...
- There is a strange issue with filtering where recurring tasks end up not coming up on the task screen if you complete a previous recurrence but the next is on the same day... -> Does this still occur?

- Link the same task to multiple notes / journals -> This is already possible.
- Recurring subtask makes allot of entries - try 'every 8 hours for 1 month' (3 entries per day...) maybe limit or allow customisation of a limit? -> There is an option to limit recurring entries to one in the future. Is there the need for more?
- Customise task completion sound -> Hm, why should the app make a noise when you check a checkbox?
- Link task to calendars already on device? (Like task.org) -> That's planned for the future and will be a bigger task
- Option to show tasks that are linked on task screen -> You can enable the flat view through the menu to show all subtasks also as standalone tasks.
- Option to do 'negative' filtering (NOT 'x' category, NOT Cancelled, etc...) -> This has been requested before, but I'm hesitating doing this in order to not make the filter screen even more stuffed
- Should cancelled tasks be considered 'completed' ? And thus not shown when filtering with 'hide completed tasks' -> I have just made this change and I'll add it in 2.5.4
- Recurring tasks to recur either by when they were due or when they were actually completed -> Also in this case I think the limit the recurring tasks to one instance in the future would be sufficient?
- Recurring tasks show every recurrence for the next year if set to 'never' end... Maybe only show the next 5 or so occurances maximum at any one point... -> Same as before...

**Comment by LoveSponge** [comment]
Hey thank you for responding, appreciate your time.

> * BUG: (known) opening item often brings blank screen -> This should hopefully be solved by now

1. Setting up a recurring task
  - daily recurrence
  - starts today
  - ends 30 days in future
2. Save
3. Tap one of the occurences
4. Edit
5. Set starts at 2 days in past
6. Save

Shows blank tasks screen on 2.5.3 beta 9.

---
> * There is a strange issue with filtering where recurring tasks end up not coming up on the task screen if you complete a previous recurrence but the next is on the same day... -> Does this still occur?

Appears this is now fixed.

---
> * Option to show tasks that are linked on task screen -> You can enable the flat view through the menu to show all subtasks also as standalone tasks.
>
> * Link the same task to multiple notes / journals -> This is already possible.

I'm unable to link the same task to 2 individual journals.

Steps:
1. Create 2 journal entries (date does not matter)
2. Create 1 task entry
3. From one of the journal screens, link the task you created
4. Go to the other journal, open the task link dialog

Task list does not include previously linked task.

I'm unable to find the option to enable the 'flat' view for tasks, could this be related to the above?

---
> * Recurring tasks to recur either by when they were due or when they were actually completed -> Also in this case I think the limit the recurring tasks to one instance in the future would be sufficient?

Not sure you understood what i meant with this one.
When take a recurring task with the following setup:

- Start today (16/08)
- Recur once a week
- Recur on START DATE

If this task is completed today or tomorrow or even next week, it will always set the next recurrence to the week after: 23/08.
If I instead setup to Recur on COMPLETION DATE, it should set the next occurance to the week after the completion date.

---
> * Recurring subtask makes allot of entries - try 'every 8 hours for 1 month' (3 entries per day...) maybe limit or allow customisation of a limit? -> There is an option to limit recurring entries to one in the future. Is there the need for more?
>
> * Recurring tasks show every recurrence for the next year if set to 'never' end... Maybe only show the next 5 or so occurances maximum at any one point... -> Same as before...

Yep fair enough i made this one twice, woops.

Where is the option to limit visible recurrences? - Looked for it on 2.5.2 and 2.5.3 beta9.

Next 1 occurrence is perfect.

---
> * Customise task completion sound -> Hm, why should the app make a noise when you check a checkbox?

This is common functionality of task list applications like M$ To Do, Tasks.org, Todoist, TickTIck, etc...
The thought behind it is that it makes you feel good for completing a task. Some task managers don't allow you to customise the sound, so it would be a big boon to this app IMO if it did.

---
> * Option to do 'negative' filtering (NOT 'x' category, NOT Cancelled, etc...) -> This has been requested before, but I'm hesitating doing this in order to not make the filter screen even more stuffed

I would maybe argue that the filtering screen could do with some work and redesign as, although it's an intuitive and easy to use implementation, not being able to negate an option or define custom relation between multiple filters (see below), leaves a hole which would basically require an extra chip just to say "do the opposite for this one" for each option.

Other apps have implemented filters/queries with more of a 'query builder' UI, which also allows you to choose the relationship between 1 filter (like 'Complete') and another ('Complete AND/OR Due date IS/IS NOT x/y/z')

Food for thought.

---
> * Should cancelled tasks be considered 'completed' ? And thus not shown when filtering with 'hide completed tasks' -> I have just made this change and I'll add it in 2.5.4

Nice!

---
Thank you again, love what you're doing with the app, really excited to see where the project is going.
